Makefile

```diff
@@ -49,4 +47,4 @@ proto.compile:
 $(eval $(RUN_ARGS):;@:)
 
 # Remove $(MAKECMDGOALS) if you don't intend make to just be a taskrunner
-.PHONY: all $(MAKECMDGOALS)
+.PHONY: all build build.develop develop run test fmt proto.compile $(MAKECMDGOALS)
```

Comment:
Thanks. I learned something and reinstated the original line :-)

Reply:
Though the original line also has a problem I think. `$(MAKECMDGOALS)` is the set of goals specified on the cmd-line, but more targets will run if any of those goals have dependencies. So `.PHONY: all $(MAKECMDGOALS)` is only sufficient if no targets in the Makefile have dependencies. If there is a dependency, it will not be .PHONY unless explicitly listed. In this case, `build.develop` should still be in there.

So adding all the targets to .PHONY isn't a bad idea IMO, but there's no reason to add them all and also have `$(MAKECMDGOALS)`.

Makefile hackery 😅

Reply:
Since this is tricky to understand -- and apparently was not understood by all developers who made updates to the `Makefile` before ;-) -- I suggest we go with explicitly listing of all goals.
However, in the current form, this still leaves the door wide open for future developers who add new goals (or remove unused ones, or rename existing ones) to overlook/forget to update the .PHONY pseudo target at the end of the Makefile. So I would prefer to “tag” each individual goal like so:

```Makefile
.PHONY: fmt
fmt:
	./scripts/fmt.sh

.PHONY: test
test:
	go test -coverprofile cover.out `go list ./...`
```

This way, the `.PHONY` "tag" would get deleted when the corresponding target gets deleted (or renamed along with renaming a target) and when new targets are added, very likely an existing target would be used as a “template” and the `.PHONY` “tag” would get copy-pasted or added for sheer “consistency” by the uninitiated

Reply:
Makefiles are fun 😅 I came across a help command that regexes double hashtag above the command, and forms a helptext https://github.com/ddelange/mapply/blob/0.1.21/Makefile

Reply:
@ddelange -- sorry I completely missed your comment. I have done a similar thing in the past as well [here](https://github.com/machine-learning-exchange/mlx/blob/main/Makefile#L9) using `grep` and `awk`

Do you want to open a PR for this, in this and the other 4 ModelMesh repos?